File: myProblems/C/2.c
// @author R. C. Howell
// @question 2
// @lang C
// @solution 4613732

/**
By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
**/

#include <stdlib.h>
#include <stdio.h>

#define MAX 4000000

// naive solution
int main(int argc, char * argv[]) {

    unsigned int total = 0;
    unsigned int currentValue = 1;
    unsigned int oldValue = 1;
    unsigned int temp;

    while (currentValue < MAX) {
        if (!(currentValue%2)) total += currentValue;
        temp = currentValue;
        currentValue += oldValue;
        oldValue = temp;
    }

    printf("%d\n", total);
    return 0;
}

File: myProblems/R/2.r
# @author R. C. Howell
# @question 2
# @lang R
# @question By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
# @solution 4613732

MAX <- 4000000

# Setup
fibs <- vector();
fibs[1] <- 1;
fibs[2] <- 2;
i <- 3;

# Create Vector
while (fibs[i-1] < MAX) { 
	fibs[i] <- sum(fibs[i-1], fibs[i-2])
	i <- i + 1
}

# Create vector of only even fib numbers
fibs <- fibs[(fibs %% 2) == 0]

sum(fibs)